integration/main_test.go

```diff
@@ -45,6 +49,7 @@ type upgradeFlowSpec struct {
 var (
 	restConfig *rest.Config
 	c          client.Client
+	pod        *corev1.Pod
```

Comment:
Avoid using package scoped variables as this has sometimes a danger of not being initialized before first use which would result into a panic.